### ISSUE TITLE
[BUG] False positive rule on tackle test app analysis with custom rule

### ISSUE THREAD

**Issue by sshveta** [konveyor/analyzer-lsp#1058]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Konveyor version

v0.9.0-alpha.2

### Priority

Major

### Current Behavior

Rule triggered : (This rule does not used to trigger earlier , also hibernate target was not selected) 
https://github.com/konveyor/rulesets/blob/62f5ff5545cb97c64a1a99e277b757611bbb9632/default/generated/eap7/110-hibernate4.windup.yaml#L995 

<img width="1871" height="1170" alt="Image" src="https://github.com/user-attachments/assets/95fc48d8-b868-4b28-bca8-e018e474be0c" />

Rule not triggered : 
Hardcoded IP Address for target cloud-readiness.

### Expected Behavior

Hardcoded IP Address for target cloud-readiness should trigger. 
hibernate4-00039 should not trigger.

### How Reproducible

Always (Default)

### Steps To Reproduce

1. Run source Analysis on tackle-testapp https://github.com/konveyor/tackle-testapp-public 
2. Target - ["Containerization", "Linux", "OracleJDK to OpenJDK"]
3. Custom rule (attached) 

Custom rule 
=====================

```
- category: mandatory
  customVariables: []
  description: Legacy configuration with io.konveyor.demo.config.ApplicationConfiguration
  effort: 1
  labels:
    - konveyor.io/source=traditional-corporate-framework
    - konveyor.io/target=cloud-corporate-framework
    - configuration
  links:
    - title: Spring Documentation - PropertySource javadoc
      url: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/PropertySource.html
    - title: Mkyong - Spring @PropertySource example
      url: https://mkyong.com/spring/spring-propertysources-example/
    - title: Baeldung - Properties with Spring and Spring Boot
      url: https://www.baeldung.com/properties-with-spring
  message: |-
    The legacy ApplicationConfiguration class is being used in this application. This is discouraged by the migration
     guidelines, and should be replaced by a more standard approach using Spring's @PropertySource annotation and Environment class:


     ```java
     @PropertySource("classpath:persistence.properties")
     public class PersistenceConfig {{
     @Autowired
     private Environment env;

     @Bean
     public DataSource dataSource() {{
     final DriverManagerDataSource dataSource = new DriverManagerDataSource();
     dataSource.setDriverClassName(env.getProperty("jdbc.driverClassName"));
     dataSource.setUrl(env.getProperty("jdbc.url"));
     dataSource.setUsername(env.getProperty("jdbc.user"));
     dataSource.setPassword(env.getProperty("jdbc.password"));

     return dataSource;
     }}
     }}
     ```

     This allows externalizing the configuration in Kubernetes by injecting it as a ConfigMap or a Secret in the lib directory from the
     container running the Tomcat instance.
  ruleID: corporate-config-01000
  when:
    java.referenced:
      location: VARIABLE_DECLARATION
      pattern: io.konveyor.demo.config.ApplicationConfiguration
```


### Environment

```markdown
- **Analyzer Image SHA** 
- **JDTLS  base image SHA**
- **Provider Settings JSON**
```

**Comment by sshveta** [comment]
Unable to upload yamls. 

Logs
==============
```
id: 8
createUser: admin
createTime: 2026-01-13T00:52:57.338938016Z
name: tackle-public-customRule_test-app-nisi176.2.analyzer
kind: analyzer
addon: analyzer
extensions:
    - java
state: Succeeded
priority: 10
data:
    mode:
        artifact: ""
        binary: false
        withDeps: false
    rules:
        labels:
            excluded: []
            included:
                - konveyor.io/target=cloud-readiness
                - konveyor.io/target=openjdk
                - konveyor.io/target=linux
                - konveyor.io/source=traditional-corporate-framework
                - konveyor.io/target=cloud-corporate-framework
                - configuration
        path: /rules
    scope:
        packages:
            excluded: []
            included: []
        withKnownLibs: false
    sources: []
    tagger:
        enabled: true
    targets: []
    verbosity: 0
application:
    id: 2
    name: tackle-public-customRule_test-app-nisi176
bucket:
    id: 13
    name: ""
started: 2026-01-13T00:52:57.573943943Z
terminated: 2026-01-13T00:54:06.846716384Z
events:
    - kind: AddonSelected
      count: 1
      last: 2026-01-13T00:52:57.531481481Z
    - kind: ExtensionSelected
      count: 1
      reason: java
      last: 2026-01-13T00:52:57.554372065Z
    - kind: PodCreated
      count: 1
      reason: konveyor-tackle/task-8-4ppb6
      last: 2026-01-13T00:52:57.606870965Z
    - kind: PodPending
      count: 2
      reason: ContainerCreating
      last: 2026-01-13T00:52:58.670523333Z
    - kind: PodRunning
      count: 62
      last: 2026-01-13T00:54:05.57247909Z
    - kind: PodSucceeded
      count: 1
      last: 2026-01-13T00:54:05.572480232Z
    - kind: ContainerKilled
      count: 1
      reason: 'container: ''java'' had not terminated.'
      last: 2026-01-13T00:54:05.744302876Z
    - kind: PodDeleted
      count: 1
      last: 2026-01-13T00:54:06.846484689Z
activity:
    - 'OptDir:    /addon/opt'
    - 'SharedDir: /shared'
    - 'CacheDir:  /cache'
    - 'SourceDir: /shared/source'
    - 'RuleDir:   /addon/rules'
    - 'BinDir:    /shared/bin'
    - 'M2Dir:     /cache/m2'
    - Fetching application.
    - '[GIT] Created: /addon/.git/19625df78bbe4a60/.gitconfig'
    - '[CMD] Running: /usr/bin/git clone https://github.com/konveyor/tackle-testapp-public /shared/source/tackle-testapp-public'
    - '[CMD] /usr/bin/git succeeded.'
    - '[RULESET] fetching: /rules'
    - '[RULESET] fetching: id=1 (.discovery)'
    - '[RULESET] fetching: id=24 (cloud-readiness)'
    - '[RULESET] fetching (dep): id=20 (.technology-usage)'
    - '[RULESET] fetching: id=23 (openjdk21)'
    - '[RULESET] fetching: id=13 (openjdk7/oraclejdk7)'
    - '[RULESET] fetching: id=15 (openjdk17/openjdk11)'
    - '[RULESET] fetching: id=14 (openjdk11/openjdk8)'
    - '[RULESET] fetching: id=17 (os/windows)'
    - '[RULESET] fetching: id=5 (eap6/java-ee/seam)'
    - '[RULESET] fetching: id=6 (eap7/weblogic/tests/data)'
    - '[RULESET] fetching: id=12 (hibernate)'
    - '[RULE] /addon/rules/files/ruleset.yaml not-found;created.'
    - '[RULE] inject(konveyor.io/include=always): /addon/rules/files/ruleset.yaml.'
    - '[CMD] Running: /usr/local/bin/konveyor-analyzer --provider-settings /addon/opt/settings.yaml --output-file /addon/insights.yaml --dep-output-file /addon/deps.yaml --no-dependency-rules --rules /addon/rules/files --rules /addon/rules/rulesets/1/rules --rules /addon/rules/rulesets/24/rules --rules /addon/rules/rulesets/20/rules --rules /addon/rules/rulesets/23/rules --rules /addon/rules/rulesets/13/rules --rules /addon/rules/rulesets/15/rules --rules /addon/rules/rulesets/14/rules --rules /addon/rules/rulesets/17/rules --rules /addon/rules/rulesets/5/rules --rules /addon/rules/rulesets/6/rules --rules /addon/rules/rulesets/12/rules --label-selector (configuration||(konveyor.io/source=traditional-corporate-framework&&(konveyor.io/target=cloud-readiness||konveyor.io/target=openjdk||konveyor.io/target=linux||konveyor.io/target=cloud-corporate-framework)))'
    - '[CMD] /usr/local/bin/konveyor-analyzer succeeded.'
    - '[TAG] Tagging Application 2.'
    - '[GIT] Created: /addon/.git/19625df78bbe4a60/.gitconfig'
    - '[CMD] Running: /usr/bin/git rev-parse HEAD'
    - '[CMD] /usr/bin/git succeeded.'
    - 'Analysis 2 reported. duration: 72.345451ms'
    - Facts updated.
    - Done.
attached:
    - id: 397
      name: git.output
      activity: 10
    - id: 400
      name: settings.yaml
      activity: 25
    - id: 401
      name: konveyor-analyzer.output
      activity: 26
    - id: 402
      name: git.output
      activity: 30
    - id: 398
      name: addon.log
    - id: 399
      name: java.log
    - id: 404
      name: pod.yaml

```
